optuna/study/study.py

```diff
@@ -632,13 +630,18 @@ def df(x):
 
         frozen_trial = self._storage.get_trial(trial_id)
 
+        if state is None and values is None:
+            self._storage.set_trial_state(trial_id, TrialState.FAIL)
+            _logger.warning("You must specify either state or values.")
+            return self._storage.get_trial(trial_id)
```

Comment:
Should this always be executed without taking into account `skip_if_finished` ? Wondering if something like the following can replace.
```suggestion
        if state is None and values is None:
            state = TrialState.FAIL
```

Reply:
The logging message might be lacking information for the user, I think adding some context would help. (Besides from the comment above)

Reply:
You suggestion is applied in adecbc4. I'll update message as well.

Reply:
I updated the doc in 6fa50b4. I'm not sure if it improves a usability so please give me a feedback. 🙏

Reply:
Thanks. Sorry for my vague comment but I meant that we could omit the warning altogether, if we get rid of the early return statement (which you did).